script/train/train_router.py

```diff
@@ -130,7 +130,7 @@ def validate_model(
             # Forward pass
             outputs = model(**inputs)
 
-            labels = labels.to(device)
+            labels = labels.to(device).float()
```

Comment:
It seems that the label is converted to long type during getitem. Maybe it will be better if we directly do the correct data conversion (long instead of float) during init?